src/lib/proof_system.ts

```diff
@@ -822,6 +846,57 @@ function dummyBase64Proof() {
   return withThreadPool(async () => Pickles.dummyBase64Proof());
 }
 
+function computeFeatureFlags(gates: Gate[]): MlFeatureFlags {
+  let flags: FeatureFlags = {
+    rangeCheck0: false,
+    rangeCheck1: false,
+    foreignFieldAdd: false,
+    foreignFieldMul: false,
+    xor: false,
+    rot: false,
+    lookup: false,
+    runtimeTables: false,
+  };
+  for (let gate of gates) {
```

Comment:
nit: This is a superficial suggestion, but you could add some type safety here instead of the switch statement

```ts
const gateTypeToFlagKey: Record<GateType, keyof FeatureFlags> = {
  RangeCheck0: 'rangeCheck0',
  RangeCheck1: 'rangeCheck1',
  ForeignFieldAdd: 'foreignFieldAdd',
  ForeignFieldMul: 'foreignFieldMul',
  Xor16: 'xor',
  Rot64: 'rot',
  Lookup: 'lookup',
  ...
};
  
for (let gate of gates) {
  const flagKey = gateTypeToFlagKey[gate.type];
  if (flagKey) {
    flags[flagKey] = true;
  }
};

Reply:
fwiw I think the switch statement is perfectly type-safe
![image](https://github.com/o1-labs/o1js/assets/20989968/4246bd2f-a72b-45be-9e57-95bde21164b8)

but your suggestion is definitely less spaghetti-like than the long switch statement

Reply:
I went with the object approach. This is the type signature you need to encode that _some but not all_ gate types cause a feature flag to be set (same as the switch statement with the default branch):
```ts
const gateToFlag: Partial<Record<GateType, keyof FeatureFlags>> = {
  RangeCheck0: 'rangeCheck0',
  RangeCheck1: 'rangeCheck1',
  ```